go/vt/vtgate/engine/limit_test.go

```diff
@@ -93,6 +146,173 @@ func TestLimitExecute(t *testing.T) {
 	}
 }
 
+func TestLimitOffsetExecute(t *testing.T) {
```

Comment:
I think it's better to add tests for those corner cases where count+offset==len, etc. That will give you the confidence on the off-by-one bugs, but will also prevent future bugs.